R/qenv-names.R

```diff
@@ -0,0 +1,33 @@
+#' The Names of a `qenv` or `qenv_error` Object
+#'
+#' Functions to get the names of a `qenv` or `qenv_error` object.
+#' The names are extrapolated from the objects in the `qenv` environment and
+#' are not stored statically, unlike the normal behavior of `names()` function.
+#'
+#' Objects named with a `.` (dot) prefix will be ignored and not returned.
+#'
+#' @param x (`qenv` or `qenv_error`) object.
+#' @return A character vector of names.
+#'
+#' @seealso [base::names()]
+#'
+#' @examples
+#' q1 <- within(qenv(), iris <- iris)
+#' names(q1)
+#'
+#' q2 <- within(q1, {
+#'   mtcars <- mtcars
+#'   CO2 <- CO2
+#' })
+#' names(q2)
+#'
+#' @export
+names.qenv <- function(x) {
+  ls(get_env(x))
+}
+
+#' @rdname names.qenv
```

Comment:
We need to decide where to put `names.qenv`. At this moment `within`, `eval_code`, `get_code` are part of `qenv` documentation and they are described in the sections. Problem with some methods (`names`, `within`) is that they have generic in base and it is not obvious that one needs to call `names.qenv`.

Reply:
If we want to keep them visible, they need to be attached to `qenv` or other docs.

Otherwise, they have to be accessed indirectly (`see also` sections or via manual search)